src/metric_model_reporter.cc

```diff
@@ -76,53 +76,123 @@ MetricModelReporter::MetricModelReporter(
   std::map<std::string, std::string> labels;
   GetMetricLabels(&labels, model_name, model_version, device, model_tags);
 
-  metric_inf_success_ =
-      CreateCounterMetric(Metrics::FamilyInferenceSuccess(), labels);
-  metric_inf_failure_ =
-      CreateCounterMetric(Metrics::FamilyInferenceFailure(), labels);
-  metric_inf_count_ =
-      CreateCounterMetric(Metrics::FamilyInferenceCount(), labels);
-  metric_inf_exec_count_ =
-      CreateCounterMetric(Metrics::FamilyInferenceExecutionCount(), labels);
-  metric_inf_request_duration_us_ =
-      CreateCounterMetric(Metrics::FamilyInferenceRequestDuration(), labels);
-  metric_inf_queue_duration_us_ =
-      CreateCounterMetric(Metrics::FamilyInferenceQueueDuration(), labels);
-  metric_inf_compute_input_duration_us_ = CreateCounterMetric(
-      Metrics::FamilyInferenceComputeInputDuration(), labels);
-  metric_inf_compute_infer_duration_us_ = CreateCounterMetric(
-      Metrics::FamilyInferenceComputeInferDuration(), labels);
-  metric_inf_compute_output_duration_us_ = CreateCounterMetric(
-      Metrics::FamilyInferenceComputeOutputDuration(), labels);
-  metric_cache_hit_count_ =
-      CreateCounterMetric(Metrics::FamilyCacheHitCount(), labels);
-  metric_cache_hit_duration_us_ =
-      CreateCounterMetric(Metrics::FamilyCacheHitDuration(), labels);
-  metric_cache_miss_count_ =
-      CreateCounterMetric(Metrics::FamilyCacheMissCount(), labels);
-  metric_cache_miss_duration_us_ =
-      CreateCounterMetric(Metrics::FamilyCacheMissDuration(), labels);
+  // Global config only for now in config map
+  auto metrics_config_map_ = Metrics::ConfigMap();
+  const auto& metrics_config_ = metrics_config_map_[""];
+
+  // Default behavior is counters for most latency metrics if no types specified
```

Comment:
I believe having MetricsConfig as a struct with all these default values initialized will be better than a standard vector of key value pair.